doc/timing.adoc

```diff
@@ -0,0 +1,259 @@
+    This is the maximum granularity (the most time spent between steps and the finest timing possible) of each game loop without a busywait.
+In singleplayer::
+    Framerate determines server steps.
+
+== Lua builtins
+
+Not restricted by mod security, these functions are available to both SSMs and CSMs and allow getting times / dates.
+
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.clock[`os.clock`] - Approximate seconds CPU time used.
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.time[`os.time`] - The current time. Will usually be a UNIX timestamp in seconds.
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.difftime[`os.difftime`] - Difference between two `os.time` timestamps (usually simply `b - a`).
+* https://www.lua.org/manual/5.1/manual.html#pdf-os.date[`os.date`] - Getting & formatting dates.
+
+=== Example
+
+You can use `os.clock` for accurate benchmarks of CPU time spent:
+
+[source,lua]
+----
+local function slow_sum(n)
+	local sum = 0
+	for i = 1, n do
+		sum = sum + i
+	end
+	return sum
+end
+
+local function benchmark(calls, func, ...)
+	local time = os.clock()
+	for _ = 1, calls do
+		func(...)
+	end
+	return (os.clock() - time) / calls -- seconds of CPU time spent per call
+end
+
+print("slow_sum(1e6) takes", benchmark(1e3, slow_sum, 1e6), "seconds per call")
+----
+
+== Functions
+
+=== `minetest.get_us_time`
+
+==== Usage
+
+[source,lua]
+----
+time = minetest.get_us_time()
+----
+
+==== Returns
+[%autowidth, frame=none]
+|===
+| `time` | integer | System-dependent timestamp in microseconds since an arbitrary starting point (µs)
+|===
+
+TIP: Divide by `1e6` to convert `time` into seconds.
+
+CAUTION: The returned `time` is not portable and not relative to any specific point in time across restarts - keep it only in memory for use while the game is running.
+
+TIP: You can use a difference of ``minetest.get_us_time``-returned times to check whether a real-world timespan has passed, which is useful for rate limiting. For in-game timers, you might prefer adding up `dtime` or - if second precision is enough - using gametime.
+
+=== Example
+
+It is possible to use a simple while-loop to wait for a timespan smaller than the server step. This is called a "busywait".
+
+[source,lua]
+----
+local start = minetest.get_us_time()
+repeat until minetest.get_us_time() - start > 1e3 -- Wait for 1000 µs to pass
+----
+
+WARNING: This blocks the server thread, possibly delaying the sending of packets that are sent each step and creating "lag". Use this only if absolutely necessary and only for very small timespans.
+
+=== `minetest.get_gametime`
+
+==== Usage
+
+[source,lua]
+----
+time = minetest.get_gametime()
+----
+
+==== Returns
+[%autowidth, frame=none]
+|===
+| `time` | integer | Time passed "in-game" since world creation in seconds ("gametime"). Does not increase while the game is paused or not running. Gametime is stored with the world and continuously increases.
```

Comment:
```suggestion
| `time` | `{type-number}` | Time passed "in-game" since world creation in seconds ("gametime"). Does not increase while the game is paused or not running. Gametime is stored with the world and continuously increases.
```